go/datas/http_batch_store.go

```diff
@@ -313,16 +332,21 @@ func resBodyReader(res *http.Response) (reader io.ReadCloser) {
 }
 
 func (bhcs *httpBatchStore) SchedulePut(c chunks.Chunk, refHeight uint64, hints types.Hints) {
-	if !bhcs.unwrittenPuts.Insert(c, refHeight) {
+	tryCache := func(c chunks.Chunk) bool {
+		bhcs.cacheMu.RLock()
+		defer bhcs.cacheMu.RUnlock()
+		return bhcs.unwrittenPuts.Insert(c)
+	}
+	if !tryCache(c) {
 		return
 	}
 
 	bhcs.requestWg.Add(1)
-	bhcs.writeQueue <- writeRequest{c.Hash(), hints, false}
+	bhcs.writeQueue <- writeRequest{hints, false}
```

Comment:
No, I think this will work. As long as  `AddHints()` uses that mutex also, I think it will work.